File: core/Params/UserHistoryParams.cs
using System;
using core.Entities.Admin;
using core.Params;

namespace core.Params
{
    public class UserHistoryParams: ParamPages
    {
          public UserHistoryParams()
          {
          }

        public UserHistoryParams(int? id, string personType, string personName, int? personId, int? applicationNo, string emailId, string mobileNo, bool createNewIfNull)
        {
            Id = id;
            PersonType = personType;
            PersonName = personName;
            PersonId = personId;
            ApplicationNo = applicationNo;
            EmailId = emailId;
            MobileNo = mobileNo;
            CreateNewIfNull = createNewIfNull;
        }

        public int? Id {get; set;}
        public int? UserHistoryHeaderId {get; set;}
        public string PersonType {get; set;}
        public string PersonName { get; set; }
        public int? PersonId {get; set;}
        public int? ApplicationNo {get; set;}
        public string EmailId {get; set;}
        public string MobileNo {get; set;}
        public DateTime DateAdded { get; set; }
        public string CategoryRef { get; set; }
        public string Status { get; set; }
        public bool? Concluded {get; set;}
        public bool CreateNewIfNull {get; set;}=false;
    }
}

File: client2/src/app/shared/params/admin/userHistoryParams.ts
export class userHistoryParams {
     id? = 0;
     userHistoryHeaderId?= 0;
     personType = '';
     personName = '';
     personId? = 0;
     customerOfficialId = 0;
     applicationNo? = 0;
     aadharNo = '';
     emailId = '';
     mobileNo = '';
     dateAdded: Date=new Date();
     categoryRef = '';
     status = '';
     concluded = null;        //default
     createNewIfNull=false;
     search = '';
     sort = '';
     pageNumber=1;
     pageSize=10;
}